### PR TITLE
findDependencies rewritten in python

### PR DESCRIPTION
**Pull request by tulamor**

### CODE REVIEW

File: SCRAM/findDependencies.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+def buildFileDeps(rel, arch, scramroot):
+  pcache = data2json("%s/.SCRAM/%s/ProjectCache.db.gz" % (rel, arch))
+  cache = {}
+  for dir in sorted(pcache["BUILDTREE"]):
+    if pcache["BUILDTREE"][dir]["SUFFIX"] != "": continue
+    if len(pcache["BUILDTREE"][dir]["METABF"]) == 0: continue
+    bf = pcache["BUILDTREE"][dir]["METABF"][0]
+    bf = re.sub(r'src\/', r'',bf)
+    if "dirs" not in cache: cache["dirs"] = {}
+    if "dir" not in cache["dirs"]: cache["dirs"][dir] = {}
```
Comment (smuzaffar) [inline SCRAM/findDependencies.py]: this shoudl be `if dir not in cache["dirs"]:`

File: SCRAM/findDependencies.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    cache["dirs"][dir] = bf
+    pack = dir
+    class_ = pcache["BUILDTREE"][dir]["CLASS"]
+    if re.search( r'^(LIBRARY|CLASSLIB)$', class_):
+      pack = pcache["BUILDTREE"][dir]["PARENT"]
+      if "packs" not in cache: cache["packs"] = {}
+      if "pack" not in cache["packs"]: cache["packs"][pack] = {}
```
Comment (smuzaffar) [inline SCRAM/findDependencies.py]: same here, check for `pack` and not `"pack"`

File: SCRAM/findDependencies.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+
+import sys, re, os, json, gzip
+from argparse import ArgumentParser
+from glob import glob
```
Comment (kreczko) [inline SCRAM/findDependencies.py]: `glob.glob` is imported but unused

**Comment by kreczko** [comment]
btw: do we have any linting for Python files within CMSSW?
flake8 reports [168 issues](https://gist.github.com/kreczko/ad1e57e6b51f443801b111cc80d9c5f9) for `SCRAM/findDependencies.py`

**Comment by smuzaffar** [comment]
@tulamor , I ran it for full cmssw and notice few things
- We need to fix couple of BuildFile.xml
  - https://github.com/cms-sw/cmssw/pull/29199
  - https://github.com/cms-sw/cmssw/pull/29200

- I see that generated py*.out files are exactly the same but rest have few differences, can you please check why we have such diff

**Comment by smuzaffar** [comment]
@tulamor , are you working any more on this? It is now months gone and we have not received any update.

**Comment by smuzaffar** [comment]
@tulamor , are you still working on it?

**Comment by tulamor** [comment]
@smuzaffar, sorry for the late reply. I am going to update this PR ASAP

**Comment by tulamor** [comment]
I have run python version of findDependencies couple of times for the full cmssw.

Realease circles used for testing:
CMSSW_11_1_X_2020-05-03-0000
CMSSW_11_1_X_2020-05-25-1100
CMSSW_11_2_X_2020-05-26-1700

- The output files are exactly the same compared with perl version (used "sort" and "diff")
- Python version runs 2x faster (used unix "time" command)

To reproduce the output:
```
scram project CMSSW_11_2_X_2020-05-26-1700
cd
cmsenv
git cms-addpkg ‘*’
git clone https://github.com/cms-sw/cmssw-config.git
python cmssw-config/SCRAM/findDependencies.py -rel `pwd` -arch $SCRAM_ARCH -scramroot /cvmfs/cms.cern.ch/share/lcg/SCRAMV1/V2_2_9_pre05/src/ 
```
Ouput location: etc/dependencies

**Comment by smuzaffar** [comment]
very well done @tulamor . I will try to recheck and include it for DEVEL IBs later.

File: SCRAM/findDependencies.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+parser.add_argument('-rel')
+parser.add_argument('-arch', dest = 'scramarch', default = os.environ.get("SCRAM_ARCH"))
+parser.add_argument('-scramroot', default = os.environ.get("SCRAMV1_ROOT"))
+args = parser.parse_args()
+
+rel = args.rel
+scramarch = args.scramarch
+scramroot = args.scramroot
+
+if scramroot is None:
```
Comment (smuzaffar) [inline SCRAM/findDependencies.py:16]: @tulamor , for python version, we do not need scramroot. so just drop it

**Comment by smuzaffar** [comment]
looks good, I have tested it for full release and I confirm that it generates correct dependency information and runs relatively faster 1m45s instead of 2m45.

**Comment by smuzaffar** [comment]
lets mergs this in master-py branch and include it in DEVEL IBs.

**Pull request merged**